src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/haflow/BaseHaFlowPathRemovalAction.java

```diff
@@ -69,7 +69,7 @@ protected void removeFlowPaths(HaPathIdsPair haPathIdsPair,
                 HaFlowPath oldPath = haFlowPathRepository.remove(haFlowPathId).orElse(null);
                 if (oldPath != null) {
                     log.debug("Removed HA-flow path {}", oldPath);
-                    HaFlowHistoryService.using(carrier).save(HaFlowHistory.withTaskId(correlationId)
+                    FlowHistoryService.using(carrier).save(HaFlowHistory.of(correlationId)
```

Comment:
Here as well, and moreover, If I got it right we have FlowHistoryService.getNextHistoryEntryTime() method for these  cases.

Reply:
there is a TODO about it. Manipulating the time value is a hack and that method shouldn't exist in the first place. The effect is probably could be seen only in test env, or rare in prod.

Reply:
I created a local variable for the flow history service.